Backend/app/dashboard/CommonAPIs/src/Dashboard/ProviderPlatform/Driver.hs

```diff
@@ -307,8 +307,31 @@ type DisableDriverAPI =
 type BlockDriverAPI =
   Capture "driverId" (Id Driver)
     :> "block"
+    :> ReqBody '[JSON] BlockDriverReq
     :> Post '[JSON] APISuccess
 
+data BlockDriverReq = BlockDriverReq
+  { code :: Text,
+    blockReason :: Maybe Text,
+    blockTimeInHours :: Maybe Int
+  }
+  deriving stock (Eq, Show, Generic)
+  deriving anyclass (ToJSON, FromJSON, ToSchema)
+
+----------------------------------------------------------
+-- block driver ----------------------------------------
+
+type DriverBlockReasonListAPI =
+  "blockReasonList"
+    :> Get '[JSON] [BlockReason]
```

Comment:
Is this list for single driver or for all drivers?
In case of single driver, you need `Capture "driverId" (Id Driver)`
In case of all drivers you need to add `driverId` in response item, also `limit` and `offset`

Reply:
no this is not dependent on driver it is a list  of reasons that we can you to block a driver , its like a termplate